app/main/services/query_builder.py

```diff
@@ -1,24 +1,36 @@
 from .conversions import strip_and_lowercase
-from ...mapping import TEXT_FIELDS, FILTER_FIELDS
+from ...mapping import TEXT_FIELDS, FILTER_FIELDS, AGGREGATED_FIELDS
 
 
-def construct_query(query_args, page_size=100):
+def construct_query(query_args, aggregations=[], page_size=100):
     if not is_filtered(query_args):
         query = {
             "query": build_keywords_query(query_args)
         }
     else:
+        query_filters = []
+        for field, values in query_args.lists():
+            if field.startswith("filter"):
+                query_filters.extend(field_filters(field, values))
+
         query = {
             "query": {
                 "filtered": {
                     "query": build_keywords_query(query_args),
                     "filter": filter_clause(query_args)
                 }
             }
-
         }
     query["highlight"] = highlight_clause()
 
+    aggregations = set(aggregations)
+    if aggregations:
+        missing_aggregations = aggregations.symmetric_difference(AGGREGATED_FIELDS)
```

Comment:
I actually think `AGGREGATABLE_FIELDS` more directly describes what it is, so I've tentatively gone with that. While `RAW_FIELDS` is accurate, it's a description of how it's implementation rather than what it does, so I think it's less apparent how/where it should be used.